Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -383,9 +383,16 @@ void ArgsParser::print_usage_markdown(FILE* file, StringView argv0)
 
 void ArgsParser::print_version(FILE* file)
 {
-    // FIXME: Allow applications to override version string for --version.
-    //        Especially useful for Lagom applications
-    outln(file, Core::Version::read_long_version_string().release_value_but_fixme_should_propagate_errors());
+    if (m_version != ""sv)
+        outln(file, m_version);
+    else
+        outln(file, Core::Version::read_long_version_string().release_value_but_fixme_should_propagate_errors());
+}
+
+void ArgsParser::set_version(StringView const& version)
```

Comment:
Regardless of the other ongoing discussion, it might be worth thinking about if we want some owning string type here instead.